kimchi/src/snarky/constraint_system.rs

```diff
@@ -1190,6 +1196,34 @@ impl<Field: PrimeField> SnarkyConstraintSystem<Field> {
                 ];
                 self.add_row(vars, GateType::Zero, vec![]);
             }
+            KimchiConstraint::Poseidon2 { states, last } => {
+                states
+                    .map(|round| round.map(|x| self.reduce_to_var(x)))
+                    .into_iter()
+                    .zip({ self.constants.poseidon.round_constants.clone() }.into_iter())
+                    .chunks(5)
+                    .into_iter()
+                    .for_each(|rounds| {
```

Comment:
I think this is pretty dense and hard to read, may I suggest spreading the logic over more lines of code to make it more readable:

```rust
let states = states.map(|round| round.map(|v| self.reduce_to_var(v));

let states_and_round_constants = states.zip(&self.constants.poseidon.round_constants);

for (round_state, round_constant) in states_and_round_constants.chunks(5) {
  // ...
}
```

```

Reply:
I thought that this approach would be easier to read than the manual indexing one, readability may be more subjective than I thought.
I'm not sure I understand the suggestion, you say to spread it over more lines to make it less dense, but your example has less lines, do you mean having more statements and intermediate variables?

Reply:
I find that style easier to read yeah, but I agree that it's suggestive. If you don't like my suggestion feel free to ignore it ;)

Reply:
I think it will be better to discuss style in a more comprehensive way and including more people, and we can maybe make some guidelines.
I will remove draft and we can revisit and change it later.